docs/specs/validation.yml

```diff
@@ -571,3 +571,427 @@ outputs:
+        "rules": [
+          {
+            "type": "MaxLevel",
+            "message": "Article URL '{0}' is '{1}' levels deep. For SEO, URLs should be '{2}' levels or fewer.",
+            "code": "path-too-deep",
+            "severity": "SUGGESTION",
+            "exclusions": [
+              "toc",
+            ],
+            "excludedDocfxVersions": [
+              "v2"
+            ]
+          }
+        ]
+      }
+    }
+  docs/folder0/a/TOC.md:
+  docs/folder0/b/TOC.json: |
+    [
+      {
+        "name":"file",
+        "href":"../file.md"
+      }
+    ]
+  docs/folder0/file.md:
+outputs:
+  docs/folder0/a/toc.json:
+  docs/folder0/b/toc.json:
+  docs/folder0/file.json:
+---
+# Content validation -  Publish Url - include toc
+inputs:
+  docfx.yml: |
+    markdownValidationRules: rules.json
+  rules.json: |
+    {
+      "publishurls": {
+        "name": "PublishUrls",
+        "description": "Validates publish urls",
+        "aliases": null,
+        "rules": [
+          {
+            "type": "MaxLevel",
+            "message": "Article URL '{0}' is '{1}' levels deep. For SEO, URLs should be '{2}' levels or fewer.",
+            "code": "path-too-deep",
+            "severity": "SUGGESTION",
+            "exclusions": [],
+            "excludedDocfxVersions": [
+              "v2"
+            ]
+          }
+        ]
+      }
+    }
+  docs/folder0/a/TOC.md:
+  docs/folder0/b/TOC.json: |
+    [
+      {
+        "name":"file",
+        "href":"../file.md"
+      }
+    ]
+  docs/folder0/file.md:
+outputs:
+  docs/folder0/a/toc.json:
+  docs/folder0/b/toc.json:
+  docs/folder0/file.json:
+  .errors.log: |
+    ["suggestion","path-too-deep","Article URL '/docs/folder0/a/toc.json' is '4' levels deep. For SEO, URLs should be '3' levels or fewer.", "docs/folder0/a/TOC.md"]
+    ["suggestion","path-too-deep","Article URL '/docs/folder0/b/toc.json' is '4' levels deep. For SEO, URLs should be '3' levels or fewer.", "docs/folder0/b/TOC.json"]
+---
+# Content validation -  Publish Url - resource
+inputs:
+  docfx.yml: |
+    markdownValidationRules: rules.json
+  rules.json: |
+    {
+      "publishurls": {
+        "name": "PublishUrls",
+        "description": "Validates publish urls",
+        "aliases": null,
+        "rules": [
+          {
+            "type": "MaxLevel",
+            "message": "Article URL '{0}' is '{1}' levels deep. For SEO, URLs should be '{2}' levels or fewer.",
+            "code": "path-too-deep",
+            "severity": "SUGGESTION",
+            "exclusions": [],
+            "excludedDocfxVersions": [
+              "v2"
+            ]
+          }
+        ]
+      }
+    }
+  docs/folder0/imagepage.md: '![](b/image.png)'
+  docs/folder0/b/image.png:
+outputs:
+  docs/folder0/b/image.png:
+  docs/folder0/imagepage.json: |
+    {"conceptual": "<p><img src=\"b/image.png\" alt=\"\" data-linktype=\"relative-path\"></p>"}
+---
+# Content validation -  Publish Url - SDP page
+inputs:
```

Comment:
you can add a second comment link to tell more details about this test, like SDP page will be not be validated